PermissionsUtilsUI/src/main/java/com/drivequant/drivekit/permissionsutils/commons/views/DiagnosisItemView.kt

```diff
@@ -0,0 +1,130 @@
+import android.support.v4.graphics.drawable.DrawableCompat
+import android.util.AttributeSet
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.LinearLayout
+import android.widget.TextView
+import com.drivequant.drivekit.common.ui.DriveKitUI
+import com.drivequant.drivekit.common.ui.extension.headLine1
+import com.drivequant.drivekit.common.ui.extension.normalText
+import com.drivequant.drivekit.common.ui.utils.DKAlertDialog
+import com.drivequant.drivekit.permissionsutils.R
+
+/**
+ * Created by Mohamed on 2020-04-16.
+ */
+// Copyright (c) 2020 DriveQuant. All rights reserved.
+
+class DiagnosisItemView : LinearLayout {
+
+    private var textViewDiagnosisTitle: TextView? = null
+    private var textViewDiagnosisSubTitle: TextView? = null
+    private var imageViewDiagnosis: ImageView? = null
+
+    private var diagnosisTitle: String? = null
+    private var diagnosisSubtitle: String? = null
+    private var diagnosisTextOK: String? = null
+    private var diagnosisTextKO: String? = null
+    private var diagnosisLink: String? = null
+
+    constructor(context: Context) : super(context) {
+        init(null)
+    }
+
+    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
+        init(attrs)
+    }
+
+    private fun init(attrs: AttributeSet?) {
+        val view = View.inflate(context, R.layout.diagnosis_item_view, null)
+        textViewDiagnosisTitle = view.findViewById(R.id.text_view_diagnosis_title)
+        textViewDiagnosisSubTitle = view.findViewById(R.id.text_view_diagnosis_subtitle)
+        imageViewDiagnosis = view.findViewById(R.id.image_view_diagnosis)
+
+        if (attrs != null) {
+            val a = context.obtainStyledAttributes(attrs, R.styleable.DiagnosisItemView, 0, 0)
+            try {
+                diagnosisTitle = a.getString(R.styleable.DiagnosisItemView_diagnosisTitle)
+                diagnosisSubtitle = a.getString(R.styleable.DiagnosisItemView_diagnosisSubtitle)
+                diagnosisTextOK = a.getString(R.styleable.DiagnosisItemView_diagnosisTextOK)
+                diagnosisTextKO = a.getString(R.styleable.DiagnosisItemView_diagnosisTextKO)
+                diagnosisLink = a.getString(R.styleable.DiagnosisItemView_diagnosisLink)
+
+                textViewDiagnosisTitle?.let {
+                    it.text = diagnosisTitle
+                    it.normalText()
+                }
+
+                textViewDiagnosisSubTitle?.let {
+                    it.text = diagnosisSubtitle
+                    it.normalText(DriveKitUI.colors.secondaryColor())
+                }
+
+            } finally {
+                a.recycle()
+            }
+        }
+
+        addView(
+            view, ViewGroup.LayoutParams(
+                ViewGroup.LayoutParams.MATCH_PARENT,
+                ViewGroup.LayoutParams.MATCH_PARENT
+            )
+        )
+    }
+
+    fun getDiagnosisTitle(): String = diagnosisTitle!!
```

Comment:
@mohamed-garci Is it possible to avoid NPE ? For example declare the variable as lateinit and not null ?

Reply:
In these case we are sure that we won't get a NPE, but yes we can declare it as a `lateinit`.

Reply:
@mohamed-garci Yes at the moment diagnosisTitle and others var are not null, but if some day we changes the code it can break that state. Moreover,  `getDiagnosisTitle()` would return String? type since there the textView that called the method will show nothing if result is null